src/coreclr/vm/jithelpers.cpp

```diff
@@ -4790,7 +4790,6 @@ HCIMPL1_RAW(Object*, JIT_CheckObj, Object* obj)
         MethodTable* pMT = obj->GetMethodTable();
         if (!pMT->ValidateWithPossibleAV()) {
             _ASSERTE(!"Bad Method Table");
-            FreeBuildDebugBreak();
```

Comment:
This debug break should stay so that this method breaks in even in release builds. Useful for testing debug JIT + release runtime combo.

Reply:
I thought that also but changed my mind when I noticed the JIT only emits this when compiled in `DEBUG`. I can instead do a fail fast?

https://github.com/dotnet/runtime/blob/8e1c9c1e9e18df1b9724f940f471b1e6f246b028/src/coreclr/jit/compiler.cpp#L4522

https://github.com/dotnet/runtime/blob/8e1c9c1e9e18df1b9724f940f471b1e6f246b028/src/coreclr/jit/importer.cpp#L17850

Reply:
Either DebugBreak() or fail fast is fine with me.

> JIT only emits this when compiled in DEBUG. 

Debug JIT does not imply debug runtime. It is common to mix and match release runtime and debug JIT.